File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/VB/WPFListBox-VB/WPFComponents/Language.vb
Public Class Language
  Private mName As String

  Public Property Name() As String
    Get
      Return mName
    End Get
    Set(ByVal value As String)
      mName = value
    End Set
  End Property

  Public Function GetLanguages() As IEnumerable(Of Language)
    Dim xe As XElement = Nothing
    Dim items As IEnumerable(Of Language) = Nothing

    Try
      xe = XElement.Load(WPFCommon.GetCurrentDirectory() & "\Xml\Languages.xml")
    Catch
      ' Ignore exceptions
    End Try

    ' Always check for null because of design time problem
    If xe IsNot Nothing Then
      items = (From item In xe.Descendants("Language") _
         Select New Language())
    End If

    Return items
  End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/Language.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;
using System.Linq.Expressions;

namespace WPFComponents
{  
  public class Language
  {
    public string Name { get; set; }

    public IEnumerable<Language> GetLanguages()
    {
      XElement xe = null;
      IEnumerable<Language> items = null;

      try
      {
        xe = XElement.Load(
          WPFCommon.GetCurrentDirectory() + 
          @"\Xml\Languages.xml");
      }
      catch
      {
        // Ignore exceptions
      }

      // Always check for null because of design time problem
      if (xe != null)
      {
        items = (from item in xe.Descendants("Language")
                 select new Language
                 {
                   Name = item.Element("Name").Value
                 });
      }

      return items;
    }
  }
}
